cmd/fdo_client/client.go

```diff
@@ -386,6 +396,31 @@ func di() (err error) { //nolint:gocyclo
 	return err
 }
 
+func getDeviceInfo() (string, error) {
+	if diDeviceInfoEnv := os.Getenv("DI_DEVICE_INFO"); len(diDeviceInfoEnv) != 0 {
+		return diDeviceInfoEnv, nil
+	}
+	if diDeviceInfo == deviceInfoSerialNumber {
+		productSerial, err := os.ReadFile("/sys/devices/virtual/dmi/id/product_serial")
```

Comment:
Reading this requires elevated permissions, which we should make note of. Perhaps write a warning if there's a read permission?

Reply:
yes, I was assuming FDO would run privileged anyway but that's a good point, will change it

Reply:
fixed